File: benchmarks/struct_sum.c
#include <stdio.h>

typedef struct {
    long long x;
    long long y;
    long long z;
} V3;

static long long sum_v3(long long n) {
    V3 acc = {0, 0, 0};
    for (long long i = 0; i < n; i++) {
        V3 v = { i, i * 2, i * 3 };
        acc.x += v.x;
        acc.y += v.y;
        acc.z += v.z;
    }
    return acc.x + acc.y + acc.z;
}

int main(void) {
    printf("%lld\n", sum_v3(10000000LL));
    return 0;
}

File: benchmarks/struct_sum.go
package main

import "fmt"

type V3 struct {
	x, y, z int64
}

func main() {
	var acc V3
	var n int64 = 10_000_000
	for i := int64(0); i < n; i++ {
		acc.x += i
		acc.y += i * 2
		acc.z += i * 3
	}
	fmt.Println(acc.x + acc.y + acc.z)
}
